### ISSUE TITLE
How to put data files where sys.argv[0] lives for onefile mode?

### ISSUE THREAD

**Issue by ThalliumSulfate** [Nuitka/Nuitka#3220]
Version:
2.5.2
Commercial: None
Python: 3.12.6 (main, Sep  8 2024, 13:18:56) [GCC 14.2.1 20240805]
Flavor: Unknown
Executable: ~/PycharmProjects/CSC/.venv/bin/python
OS: Linux
Arch: x86_64
Distribution: Manjaro (based on arch) None
Version C compiler: /usr/bin/gcc (gcc 14.2.1).

Installed Via Pip in a virtualenv

 PyPI Packages:
PyQt6 - 6.7.1 
PyQt6_sip - 13.8.0 
Datafiles - 2.3

File Stucture:
```
package/
          main.py
          path.py
          Data/
               *.yml
          src/
               data_management.py
               main_gui.py
               ability.py
               ancestry.py
               character.py
```

Code Snippets:

My Path file:
```
import os
import sys

filePath = os.path.join(os.path.dirname(__file__))
appPath = os.path.join(os.path.dirname(sys.argv[0]))
```

Data management module:
```
def loadConstantData():
    dPath = filePath + '/data/abilities'

    directories = next(walk(dPath))[1]

    for dir in directories:
        files = next(walk(os.path.join(dPath, dir)), (None, None, []))[2]
        for file in files:
            restored = Ability(file.replace(".yml", ""), dir, Missing, Missing)
            match dir:
                case 'natural':
                    naturalAbilityList.append(restored)
                case 'perk':
                    perkAbilityList.append(restored)

    dPath = filePath + '/data/ancestries'


    files = next(walk(dPath), (None, None, []))[2]
    for file in files:
        restored = Ancestry(file.replace(".yml", ""), Missing)
        ancestryList.append(restored)

def hasData():
    data = False
    dPath = appPath + '/characters/'

    if not os.path.isdir(dPath):
       os.mkdir(dPath)

    filenames = next(walk(dPath), (None, None, []))[2]

    if not len(filenames) == 0:
        data = True
        characterList.clear()
        for fileName in filenames:
            restored = Character(fileName.replace(".yml", ""), Missing, Missing)
            characterList.append(restored)

    return data
```

When running:
`print(appPath)`
The path is the same location as
`print(filePath)`
which means I cannot modify saved data(this is clearly fine in standalone, however not convenient for my client)

Options Used:
--onefile
--include-data-dir={MAIN_DIRECTORY}/data=data 
--plugin-enable=pyqt6

**Discussion**
* The hasData() is for saved data at the original compiled file location

* loadConstantData() is for loading pre-made DataFiles 

* loadConstantData() runs as expected since the __file__ path leads to the correct temp folder

* These both run at launch

* This has also been tried with --standalone, can modify and generate saved files, however it generates inside the standalone folder

**Comment by kayhayen** [comment]
If you want files to live near the onefile executable, that is what `--include-onefile-external-data` is for, what adding `--include-onefile-external-data=data` will make it live on the outside.

This option, despite it's name works for --standalone too, but there you ought to refer to its location via `__compiled__.containing_dir` which abstracts outside for standalone, app bundle, and onefile, but if you only use onefile, then `dirname(sys.argv[0])` has the big advantage of remaining portable code.

**Comment by ThalliumSulfate** [comment]
> If you want files to live near the onefile executable, that is what `--include-onefile-external-data` is for, what adding `--include-onefile-external-data=data` will make it live on the outside.
> 
> This option, despite it's name works for --standalone too, but there you ought to refer to its location via `__compiled__.containing_dir` which abstracts outside for standalone, app bundle, and onefile, but if you only use onefile, then `dirname(sys.argv[0])` has the big advantage of remaining portable code.

ah, it seems like there was a misunderstanding, The major issue that I'm facing is sys.argv[0] is leading to the temp folder not the compiled directory. 

I can include the external data, however I will not be able to access it as the dirname(sys.argv[0]) paths to my /tmp

**Comment by kayhayen** [comment]
That should be so wrong, but maybe I added a regression there recently, my bad, it's supposed to point to the original executable.

**Comment by kayhayen** [comment]
There will be an emergency hotfix for this breaks a lot of use cases naturally.

**Comment by kayhayen** [comment]
This should be fixed in 2.5.4 hotfix release that I just made, thanks for persisting in your report, I didn't get it at all. Obviously, we lack test coverage for the option I mentioned to make data external that demonstrates successful usage of it from there.
